controllers/update_pods.go

```diff
@@ -47,6 +48,7 @@ func (updatePods) reconcile(ctx context.Context, r *FoundationDBClusterReconcile
 	}
 
 	updates := make(map[string][]*corev1.Pod)
+	zoneProcessGroupMap := make(map[string][]string)
```

Comment:
I don't think we need this additional struct, all Pods will have the important information.

Reply:
Any comment on this? It seems a little bit wasteful to generate this for all process groups instead of only the ones that are selected for removal. I'm okay with this as an initial implementation be we should have a follow up to improve this.